crates/sui-json-rpc-api/src/rpc_types.rs

```diff
@@ -1292,3 +1292,24 @@ impl From<TypeTag> for SuiTypeTag {
         Self(format!("{}", tag))
     }
 }
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub enum RPCTransactionRequestParams {
+    TransferCoinRequestParams(RPCTransferCoinRequestParams),
+    MoveCallRequestParams(RPCMoveCallRequestParams),
```

Comment:
nit: can we rename these to `TransferCoin` and `MoveCall` ? the json request for `batchTransaction` looks like this now, will be good if we can shorten it a bit.
```
{
    "jsonrpc": "2.0",
    "method": "sui_batchTransaction",
    "params": [
        "0x1f6a998fe146d946e843c5c4aeeca44731195e42",
        [{"TransferCoinRequestParams": {"recipient": "0x1f6a998fe146d946e843c5c4aeeca44731195e42","object_id": "0xca018e48cf9f179fabc39406beeaf82305163715"}}],
        null,
        1000
    ],
    "id": 1
}
```